File: emedia/src/main/java/com/xjl/emedia/utils/Logger.java
package com.xjl.emedia.utils;

import android.util.Log;

/**
 * Created by x33664 on 2019/1/29.
 */

public class Logger {

    private static final String TAG = "EMedia";

    private static boolean openLogger = true;

    public static void isOpen(boolean open) {
        openLogger = open;
    }

    public static void d(String content) {
        if (openLogger)
            Log.d(TAG, content);
    }

    public static void i(String content) {
        if (openLogger)
            Log.i(TAG, content);
    }

    public static void e(String content) {
        if (openLogger)
            Log.e(TAG, content);
    }

}

File: emedia/src/main/java/com/xjl/emedia/logger/Logger.kt
package com.xjl.emedia.logger


/**
 * Created by x33664 on 2019/1/29.
 */
object Logger {
    private const val TAG = "EMedia"
    private var openLogger = true
    private var loggerInterface: LoggerInterface = DefaultLogger()

    @JvmStatic
    fun isOpen(open: Boolean) {
        openLogger = open
    }

    @JvmStatic
    fun setLogger(loggerInterface: LoggerInterface) {
        this.loggerInterface = loggerInterface
    }

    @JvmStatic
    fun d(content: String) {
        if (openLogger) loggerInterface.d(TAG, content)
    }

    @JvmStatic
    fun i(content: String) {
        if (openLogger) loggerInterface.i(TAG, content)
    }

    @JvmStatic
    fun w(content: String) {
        if (openLogger) loggerInterface.w(TAG, content)
    }

    @JvmStatic
    fun e(content: String) {
        if (openLogger) loggerInterface.e(TAG, content)
    }
}
